src/main/java/com/minecolonies/apiimp/initializer/InteractionValidatorInitializer.java

```diff
@@ -175,12 +175,19 @@ public static void init()
           });
 
         InteractionValidatorRegistry.registerStandardPredicate(new TranslationTextComponent(FURNACE_USER_NO_ORE),
-          citizen -> citizen.getWorkBuilding() instanceof BuildingSmeltery && IColonyManager.getInstance().getCompatibilityManager()
-                                                                                .getSmeltableOres()
-                                                                                .stream()
-                                                                                .anyMatch(storage -> !((BuildingSmeltery) citizen.getWorkBuilding()).getCopyOfAllowedItems()
-                                                                                                        .getOrDefault(ORE_LIST, new ArrayList<>())
-                                                                                                        .contains(storage)));
+          citizen -> {
+            if (citizen.getWorkBuilding() instanceof BuildingSmeltery)
+            {
+                final List<ItemStorage> oreList = ((BuildingSmeltery) citizen.getWorkBuilding()).getModuleMatching(ItemListModule.class, m -> m.getId().equals(ORE_LIST))
+                                                    .map(ItemListModule::getList).orElse(ImmutableList.of());
+                return IColonyManager.getInstance().getCompatibilityManager()
+                  .getSmeltableOres()
+                  .stream()
+                  .anyMatch(storage -> !oreList.contains(storage));
+            }
+            return false;
```

Comment:
I would say this is hardly in the scope of this PR. I'm just querying the module instead of the abstract class.
The reason we check the instance is just that if the job changed, this is not valid anymore.

Reply:
This is just as much inside of the scope of this PR as anything else in the PR.
This changes the base functionality of the underlying implementation, as such it is your task as the creator of the PR to also provide methods that help with readability and understandability of the code, which this suggested change targets.

From this perspective it makes a lot of sense to continue the line of the functional programming down into this piece of code here, and as such make it better readable.